unlock-app/src/middlewares/walletMiddleware.js

```diff
@@ -53,6 +54,13 @@ export default function walletMiddleware({ getState, dispatch }) {
   })
 
   walletService.on('transaction.new', (transactionHash, from, to) => {
+    // If we're waiting (full-screen wallet notification overlay is active), we
+    // can safely dismiss it now
+    const { waiting } = getState().walletStatus
```

Comment:
Shouldn't that check happen in the reducer?

Reply:
Great point!